### ISSUE TITLE
Interface Array runtime bugs 

### ISSUE THREAD

**Issue by kahsa** [vlang/v#7123]
Is this known issue?

```v
import gg

struct Parent {
mut:
	children []Child
}

interface Child {
	add(int)
}

struct Taro {
mut:
	items []int
}

fn (mut s Taro) add(i int) {
	s.items << i
}

fn (mut p Parent) set() {
	mut taro := Taro{}
	taro.add(123)
	taro.add(456)
	taro.add(789)
	for i in taro.items {
		println('Taro added $i')     // line 27
	}
	p.children << taro
}

// ver.1 -------------------------
/*
fn main() {
	mut parent := Parent{}
	parent.set()
	child := parent.children[0]
	if child is Taro {
		println('Taro has $child.items.len items, first is ${child.items[0]}')
	}
}
*/
// ver.2 -------------------------
fn main() {
	mut parent := Parent{}
	parent.set()
	child := parent.children[0]
	show(parent)
}

fn show(parent Parent) {
	child := parent.children[0]
	if child is Taro {
		println('Taro has $child.items.len items, first is ${child.items[0]}')
	}
}
// ver.3 -------------------------
/*
struct App {
mut:
	parent Parent
	gg     &gg.Context
	count  int
}

fn (mut a App) draw() {
	if a.count > 0 {
		return
	}
	child := a.parent.children[0]
	if child is Taro {
		println("Taro\'s items len is $child.items.len, first of it is ${child.items[0]}")
	}
	a.count++
}

fn frame(mut app App) {
	app.gg.begin()
	app.draw()
	app.gg.end()
}

// BUG main()外の関数で生成したInterefaceのArrayがThreadで別物になる
fn main() {
	mut app := App{
		parent: Parent{}
		gg: 0
	}
	app.gg = gg.new_context({
		width: 300
		height: 200
		use_ortho: true
		create_window: true
		frame_fn: frame
		user_data: &app
	})
	app.parent.set()
	app.gg.run()
}
*/
```
ver.1 prints
```
Taro added 123
Taro added 456
Taro added 789
Taro has 3 items, first is 123
```
This works as expected.

ver.2 prints
```
Taro added 123
Taro added 456
Taro added 789
Taro has 1 items, first is -953793120
```
items count and items[0] value both missed.

ver.3 prints
```
Taro added 123
Taro added 456
Taro added 789
Taro's items len is 3, first of it is 184
```
items count is ok but it's value is incorrect.
and I wonder where 184 comes from.
it is always 184

-----------------------------------------------

And when uncommented line 26 - 28
(I heared this is a well known issue)
```
26	// for i in taro.items {
27	// 	println('Taro added $i')
28	// }
```
ver.1 prints
```
Terminated by signal 139 (SIGSEGV)
```

ver.2 prints the same
```
Terminated by signal 139 (SIGSEGV)
```

ver.3 prints ...
```
Taro's items len is 3, first of it is 0
```
right count, wrong value.

------------------------------------------------
And now uncomment line 27 only
```
26	for i in taro.items {
27		// println('Taro added $i')
28	}
```

ver.1 prints
```
Taro has 3 items, first is 123
```
correct.

ver.2 prints
```
V panic: array.get: index out of range (i == 0, a.len == -1111821888)
                                               | 0x55b3439b06a8 | /home/tsukasa/develop/vprojects/mytradetool/learn/bugs/lostarr(+0x406a8) 
                                               | 0x55b3439d07d6 | /home/tsukasa/develop/vprojects/mytradetool/learn/bugs/lostarr(+0x607d6) 
                                               | 0x55b3439d0751 | /home/tsukasa/develop/vprojects/mytradetool/learn/bugs/lostarr(+0x60751) 
                                               | 0x55b3439d8a28 | /home/tsukasa/develop/vprojects/mytradetool/learn/bugs/lostarr(+0x68a28) 
                                               | 0x7f62c489a152 | /usr/lib/libc.so.6(__libc_start_main+0xf2) 
                                               | 0x55b3439760ce | /home/tsukasa/develop/vprojects/mytradetool/learn/bugs/lostarr(+0x60ce) 
Terminated by signal 139 (SIGSEGV)
```

ver.3 prints
```
V panic: array.get: index out of range (i == 0, a.len == -1809950688)
                                               | 0x56061209e6a8 | /home/tsukasa/develop/vprojects/mytradetool/learn/bugs/lostarr(+0x406a8) 
                                               | 0x5606120be757 | /home/tsukasa/develop/vprojects/mytradetool/learn/bugs/lostarr(+0x60757) 
                                               | 0x5606120be7d4 | /home/tsukasa/develop/vprojects/mytradetool/learn/bugs/lostarr(+0x607d4) 
                                               | 0x5606120b9457 | /home/tsukasa/develop/vprojects/mytradetool/learn/bugs/lostarr(+0x5b457) 
                                               | 0x560612078e85 | /home/tsukasa/develop/vprojects/mytradetool/learn/bugs/lostarr(+0x1ae85) 
                                               | 0x5606120795fe | /home/tsukasa/develop/vprojects/mytradetool/learn/bugs/lostarr(+0x1b5fe) 
                                               | 0x56061207c3ff | /home/tsukasa/develop/vprojects/mytradetool/learn/bugs/lostarr(+0x1e3ff) 
                                               | 0x56061207c4d2 | /home/tsukasa/develop/vprojects/mytradetool/learn/bugs/lostarr(+0x1e4d2) 
                                               | 0x5606120b5945 | /home/tsukasa/develop/vprojects/mytradetool/learn/bugs/lostarr(+0x57945) 
                                               | 0x5606120b9b66 | /home/tsukasa/develop/vprojects/mytradetool/learn/bugs/lostarr(+0x5bb66) 
                                               | 0x5606120be942 | /home/tsukasa/develop/vprojects/mytradetool/learn/bugs/lostarr(+0x60942) 
                                               | 0x5606120c6b69 | /home/tsukasa/develop/vprojects/mytradetool/learn/bugs/lostarr(+0x68b69) 
                                               | 0x7f4902f68152 | /usr/lib/libc.so.6(__libc_start_main+0xf2) 
                                               | 0x5606120640ce | /home/tsukasa/develop/vprojects/mytradetool/learn/bugs/lostarr(+0x60ce) 
Terminated by signal 139 (SIGSEGV)
```

--------------------------------------------------
`v doctor` is 
```
OS: linux, Linux version 5.9.11-arch2-1 (linux@archlinux) (gcc (GCC) 10.2.0, GNU ld (GNU Binutils) 2.35.1) #1 SMP PREEMPT Sat, 28 Nov 2020 02:07:22 +0000
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i5-5250U CPU @ 1.60GHz
CC version: cc (GCC) 10.2.0

getwd: /home/tsukasa/develop/vprojects/mytradetool
vroot: /home/tsukasa/lang/v
vexe: /home/tsukasa/lang/v/v
vexe mtime: 2020-12-04 01:44:53
is vroot writable: true
V full version: V 0.1.30 1ead130.d12f5f7

Git version: git version 2.29.2
Git vroot status: weekly.2020.49.3-3-gd12f5f7b (4 commit(s) behind V master)
.git/config present: true
thirdparty/tcc: N/A
```

I need a workaround ver.3 for now.

**Comment by kahsa** [comment]
I want to know how to work around these for now.

**Comment by kahsa** [comment]
a little bit added
```v
// ver.2 
fn main() {
	mut parent := Parent{}
	parent.set()
	child := parent.children[0]
	show(parent)
	//added below
	child2 := parent.children[0]
	if child2 is Taro {
		println('Taro has $child2.items.len items, first is ${child2.items[0]}')
	}
}

fn show(parent Parent) {
	child := parent.children[0]
	if child is Taro {
		println('Taro has $child.items.len items, first is ${child.items[0]}')
	}
}
```
I expected the second println() prints same to first, but ..
```
Taro added 123
Taro added 456
Taro added 789
Taro has 1 items, first is -1409507328
V panic: array.get: index out of range (i == 0, a.len == -1409507248)
                                               | 0x5650636a46a8 | /home/tsukasa/develop/vprojects/mytradetool/learn/bugs/lostarr(+0x406a8) 
                                               | 0x5650636c47e6 | /home/tsukasa/develop/vprojects/mytradetool/learn/bugs/lostarr(+0x607e6) 
                                               | 0x5650636ccaf8 | /home/tsukasa/develop/vprojects/mytradetool/learn/bugs/lostarr(+0x68af8) 
                                               | 0x7fc7b8654152 | /usr/lib/libc.so.6(__libc_start_main+0xf2) 
                                               | 0x56506366a0ce | /home/tsukasa/develop/vprojects/mytradetool/learn/bugs/lostarr(+0x60ce) 
Terminated by signal 139 (SIGSEGV)
```

**Comment by kahsa** [comment]
fixed by #10529
